File: tools/web/goshop/js/service/contentCategoryService.js
//服务层
var contentCategoryUrl='http://localhost:8999/shopping-content-ms/contentCategory';
app.service('contentCategoryService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get(contentCategoryUrl+'/findAll');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get(contentCategoryUrl+'/findPage?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get(contentCategoryUrl+'/findOne?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post(contentCategoryUrl+'/add',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post(contentCategoryUrl+'/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get(contentCategoryUrl+'/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post(contentCategoryUrl+'/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});

File: shopping-parents/shopping-content-ms/src/main/java/com/online/shopping/service/ContentCategoryService.java
package com.online.shopping.service;

import java.util.List;

import com.online.shopping.entity.PageResult;
import com.online.shopping.pojo.TbContentCategory;


/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface ContentCategoryService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbContentCategory> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(TbContentCategory tbContentCategory);
	
	
	/**
	 * 修改
	 */
	public void update(TbContentCategory tbContentCategory);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbContentCategory findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

}
